crates/consensus/src/signed.rs

```diff
@@ -59,22 +59,22 @@ impl<T: Transaction> Signed<T> {
     }
 }
 
-impl<T: Transaction> alloy_rlp::Encodable for Signed<T> {
+impl<T: SignableTransaction<Sig>, Sig> alloy_rlp::Encodable for Signed<T, Sig> {
```

Comment:
i forget why we did Sig as a generic instead of associated type, maybe a follow-up breaking change to the trait could remove the double mention of `Sig` to make the `Signed` trait easier to impl

Reply:
because transactions can be signed with multiple types of signatures on some networks